File: 位运算/1310.cpp
#include<vector>
using namespace std;
class Solution {
public:
    vector<int> xor_arrQueries(vector<int>& arr, vector<vector<int>>& queries) {
        vector<int>xor_arr;
        xor_arr.push_back(0);
        for(auto a:arr)
            xor_arr.push_back(xor_arr.back()^a);
        vector<int>res;
        for(auto &q:queries)
            res.push_back(xor_arr[q[0]]^xor_arr[q[1]+1]);
        return res;
    }
};

File: 位运算/1310.py
from typing import List


class Solution:
    def xorQueries(self, arr: List[int], queries: List[List[int]]) -> List[int]:
        '''暴力法O(nm)超时,采用前缀数组'''
        xor = [0]
        for a in arr:
            xor.append(xor[-1]^a)

        res = []
        for l,r in queries:
            res.append(xor[l]^xor[r+1])
        return res


arr =  [4,8,2,10]
queries = [[2,3],[1,3],[0,0],[0,3]]
obj = Solution()
obj.xorQueries(arr,queries)